packit_service/models.py

```diff
@@ -920,15 +921,23 @@ class SRPMBuildModel(Base):
     # our logs we want to show to the user
     logs = Column(Text)
     success = Column(Boolean)
+    job_trigger_id = Column(Integer, ForeignKey("build_triggers.id"))
+    job_trigger = relationship("JobTriggerModel", back_populates="srpm_builds")
     copr_builds = relationship("CoprBuildModel", back_populates="srpm_build")
     koji_builds = relationship("KojiBuildModel", back_populates="srpm_build")
 
     @classmethod
-    def create(cls, logs: str, success: bool) -> "SRPMBuildModel":
+    def create(
+        cls, logs: str, success: bool, trigger_model: AbstractTriggerDbType,
+    ) -> "SRPMBuildModel":
+        job_trigger = JobTriggerModel.get_or_create(
+            type=trigger_model.job_trigger_model_type, trigger_id=trigger_model.id,
+        )
         with get_sa_session() as session:
             srpm_build = cls()
             srpm_build.logs = logs
             srpm_build.success = success
+            srpm_build.job_trigger = job_trigger
             session.add(srpm_build)
             return srpm_build
 
```

Comment:
Should this be converted to get_or_create style or is only create fine ?
If so, how and why?

Reply:
@TomasTomecek ping :smiley:

Reply:
I am used to django's API, hence the get_or_create methods; this project is the first time I work with flask/SA more.

If it's needed somewhere in the code, feel free to implement get_or_create, if not, we can have `create` only, that's fine. Since we don't update them, afaik, create should be perfectly fine.